### PR TITLE
[NTUSER] window.c: Don't use RETURN() macro

### PR DESCRIPTION
**Pull request by TAN-Gaming**
## Proposed changes
- Don't use RETURN() macro
- Remove parentheses around return value

### CODE REVIEW

File: win32ss/user/ntuser/window.c [hunk 1]
```diff
@@ -3704,13 +3692,13 @@ HWND FASTCALL UserGetShellWindow(VOID)
    if (!NT_SUCCESS(Status))
    {
       SetLastNtError(Status);
-      return( (HWND)0);
+      return (HWND)0;
```
Comment (HBelusca) [inline win32ss/user/ntuser/window.c:3695]:
```suggestion
      return NULL;
```

File: win32ss/user/ntuser/window.c [hunk 2]
```diff
@@ -3418,25 +3410,23 @@ HWND APIENTRY
...
-      RETURN(NULL);
-   }
+      Ancestor = UserGetAncestor(Window, Type);
+      /* fixme: can UserGetAncestor ever return NULL for a valid window? */
 
-   Ancestor = UserGetAncestor(Window, Type);
-   /* faxme: can UserGetAncestor ever return NULL for a valid window? */
-
-   RETURN(Ancestor ? Ancestor->head.h : NULL);
+      Ret = (Ancestor ? Ancestor->head.h : NULL);
```
Comment (HBelusca) [inline win32ss/user/ntuser/window.c:3424]:
```suggestion
      Ret = (Ancestor ? UserHMGetHandle(Ancestor) : NULL);
```

File: win32ss/user/ntuser/window.c [hunk 3]
```diff
@@ -3499,20 +3489,23 @@ NtUserGetComboBoxInfo(
...
-      RETURN(FALSE);
+      Ret = FALSE;
+      goto Exit;
    }
 
    // Pass the user pointer, it was already probed.
    if ((Wnd->pcls->atomClassName != gpsi->atomSysClass[ICLS_COMBOBOX]) && Wnd->fnid != FNID_COMBOBOX)
    {
-      RETURN( (BOOL) co_IntSendMessage( Wnd->head.h, CB_GETCOMBOBOXINFO, 0, (LPARAM)pcbi));
+      Ret = (BOOL)co_IntSendMessage(Wnd->head.h, CB_GETCOMBOBOXINFO, 0, (LPARAM)pcbi);
```
Comment (HBelusca) [inline win32ss/user/ntuser/window.c:3507]:
```suggestion
      Ret = (BOOL)co_IntSendMessage(UserHMGetHandle(Wnd), CB_GETCOMBOBOXINFO, 0, (LPARAM)pcbi);
```

File: win32ss/user/ntuser/window.c [hunk 4]
```diff
@@ -3592,19 +3583,19 @@ NtUserGetListBoxInfo(
...
    if (!(Wnd = UserGetWindowObject(hWnd)))
    {
-      RETURN( 0 );
+      goto Exit;
    }
 
    if ((Wnd->pcls->atomClassName != gpsi->atomSysClass[ICLS_LISTBOX]) && Wnd->fnid != FNID_LISTBOX)
    {
-      RETURN( (DWORD) co_IntSendMessage( Wnd->head.h, LB_GETLISTBOXINFO, 0, 0 ));
+      Ret = (DWORD)co_IntSendMessage(Wnd->head.h, LB_GETLISTBOXINFO, 0, 0);
```
Comment (HBelusca) [inline win32ss/user/ntuser/window.c:3597]:
```suggestion
      Ret = (DWORD)co_IntSendMessage(UserHMGetHandle(Wnd), LB_GETLISTBOXINFO, 0, 0);
```

**Review by HBelusca — COMMENTED** [review]
looks good. Suggestions to use the UserHMGetHandle macro instead of digging into the head thing.

**Review by HBelusca — APPROVED** [review]

**Comment by JoachimHenze** [comment]
What was the RETURN macro initially invented for?

**Comment by TAN-Gaming** [comment]
> What was the RETURN macro initially invented for?

It's just a wrapper for goto.

[https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/user/ntuser/ntuser.h;h=7c878c4f793b20432cdf370f84c48a713d520aab;hb=HEAD#l4](https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/user/ntuser/ntuser.h;h=7c878c4f793b20432cdf370f84c48a713d520aab;hb=HEAD#l4)

**Review by katahiromz — APPROVED** [review]

**Review by JoachimHenze — APPROVED** [review]

**Comment by JoachimHenze** [comment]
VBox https://reactos.org/testman/compare.php?ids=91926,91929
KVM https://reactos.org/testman/compare.php?ids=91925,91930 (+2 reboots)

KVM run looks terrible, but finding out whether it is related to the change requires more runs. My gutfeeling points to that being unrelated.

**Pull request merged**